src/test/java/com/artipie/front/settings/RepoDataTest.java

```diff
@@ -81,6 +89,104 @@ void movesData() {
         );
     }
 
+    @ParameterizedTest
+    @ValueSource(
+        strings = {"_storage.yaml", "_storage.yml", "my-repo/_storage.yaml", "my-repo/_storage.yml"}
+    )
+    void movesDataWithAliasAndFlatLayout(final String key) {
+        this.stngs.save(
+            new Key.From(String.format("%s.yml", RepoDataTest.REPO)),
+            this.repoSettingsWithAlias().getBytes(StandardCharsets.UTF_8)
+        );
+        this.stngs.save(
+            new Key.From(key),
+            this.storageAlias().getBytes(StandardCharsets.UTF_8)
+        );
+        this.data.save(new Key.From(RepoDataTest.REPO, "first.txt"), new byte[]{});
+        this.data.save(new Key.From(RepoDataTest.REPO, "second.txt"), new byte[]{});
+        final String nrepo = "new-repo";
+        new RepoData(new RepoSettings("flat", this.stngs))
+            .move(RepoDataTest.REPO, "any", nrepo).toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            "Repository data are moved",
+            this.data.list(Key.ROOT).stream().map(Key::string).collect(Collectors.toList()),
```

Comment:
Should we check that the old repository data doesn't exist?

Reply:
same here